### PR TITLE
Add ideation payload to chain

### PR DESCRIPTION
**Pull request by dnadales**
I tried to distill some of @nkarag ideas into a simpler PR where we can discuss first how to incorporate the ideation payload to the chain. At the moment the signal generator produce invalid signals. I have to find out what's wrong, but this gives at least a base to discuss.

### CODE REVIEW

File: executable-spec/src/Cardano/Ledger/Spec/STS/Chain/Transactions.hs [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+data Env =
+  Env { currentSlot :: Slot
+      , participants :: Bimap Core.VKey Core.SKey
+      }
+  deriving (Eq, Show)
+
+
+data St =
+  St { dummySt :: State Dummy.TRANSACTION
+     , ideationSt :: State IDEATION
```
Comment (nkarag) [inline executable-spec/src/Cardano/Ledger/Spec/STS/Chain/Transactions.hs]: We do not store some state that can give us the slot of an Ideation Signal. We only have a `Map` from Dummy Transactions to Slots. We need to be able to get fast, the slot of a specific update event, in order to do slot arithmetic.

File: executable-spec/src/Cardano/Ledger/Spec/STS/Chain/Transactions.hs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+                        & takeWhile ((< maximumSize - 5) . snd)
+                        & fmap fst
+      where
+        -- We compute the cumulative sum of the transaction sizes. We add 3 to
+        -- account for the list constructor.
+        sizes :: [WordCount]
+        sizes = scanl (\acc tx -> acc + size tx + 3) 0 txs
+
+    transactionGen _ (Env { participants }) (St { ideationSt }) =
+      Gen.frequency [ (9, Dummy <$> Dummy.genTransaction)
```
Comment (nkarag) [inline executable-spec/src/Cardano/Ledger/Spec/STS/Chain/Transactions.hs]: With these frequencies we can have empty Blocks generated (i.e., with no transactions at all). Do we want this?
We also generate slot numbers that are not consecutive. Therefore we have two "sources" for generating empty blocks.

File: executable-spec/src/Cardano/Ledger/Spec/STS/Chain/Transactions.hs (hunk 1 above)
Comment (nkarag) [reply executable-spec/src/Cardano/Ledger/Spec/STS/Chain/Transactions.hs]: Maybe we need to add in the Ideation state the following
```haskell
Map (Signal IDEATION) Slot
```

**Comment by dnadales** [comment]
Merging this in spite of [strange buildkite failure](https://buildkite.com/input-output-hk/decentralized-software-updates/builds/36#8fb5777a-02d7-4678-bcb9-c484d9a44d87/34-40), since the relevant checks (building and testing) are passing.

**Pull request merged**